src/event-log/event-emitter-stream.ts

```diff
@@ -0,0 +1,53 @@
+import type { GenericMessage } from '../types/message-types.js';
+import type { KeyValues } from '../types/query-types.js';
+import type { EventListener, EventStream, EventSubscription } from '../types/subscriptions.js';
+
+import { EventEmitter } from 'events';
+
+const EVENTS_LISTENER_CHANNEL = 'events';
+
+type EventStreamEmitterConfig = {
+  emitter?: EventEmitter;
+};
+
+export class EventEmitterStream implements EventStream {
+  private eventEmitter: EventEmitter;
+  private isOpen: boolean = false;
+
+  constructor(config?: EventStreamEmitterConfig) {
```

Comment:
For my understanding: what's the reason for needing to pass and `EventEmitter` as opposed to just create it for the caller?